packages/@aws-cdk/aws-stepfunctions-tasks/lib/emrcontainers/start-job-run.ts

```diff
@@ -0,0 +1,685 @@
+      new iam.PolicyStatement({
+        resources: [
+          cdk.Stack.of(this).formatArn({
+            service: 'eks',
+            resource: 'cluster',
+            resourceName: eksClusterInfo.getResponseField('virtualCluster.containerProvider.id'),
+          }),
+        ],
+        actions: [
+          'eks:DescribeCluster',
+        ],
+      }),
+    );
+    shellCliLambda.addToRolePolicy(
+      new iam.PolicyStatement({
+        resources: [role.roleArn],
+        actions: [
+          'iam:GetRole',
+          'iam:UpdateAssumeRolePolicy',
+        ],
+      }),
+    );
+    const provider = new cr.Provider(this, 'CustomResourceProvider', {
+      onEventHandler: shellCliLambda,
+    });
+    new cdk.CustomResource(this, 'Custom Resource', {
+      serviceToken: provider.serviceToken,
+    });
+  }
+
+  private createPolicyStatements(): iam.PolicyStatement[] {
+    const policyStatements = [
+      new iam.PolicyStatement({
+        resources: [
+          cdk.Stack.of(this).formatArn({
+            arnFormat: cdk.ArnFormat.SLASH_RESOURCE_SLASH_RESOURCE_NAME,
+            service: 'emr-containers',
+            resource: 'virtualclusters',
+            resourceName: sfn.JsonPath.isEncodedJsonPath(this.props.virtualCluster.id) ? '*' : this.props.virtualCluster.id, // Need wild card for dynamic start job run https://docs.aws.amazon.com/step-functions/latest/dg/emr-eks-iam.html
+          }),
+        ],
+        actions: ['emr-containers:StartJobRun'],
+        conditions: {
+          StringEquals: {
+            'emr-containers:ExecutionRoleArn': this.role.roleArn,
+          },
+        },
+      }),
+    ];
+
+    if (this.integrationPattern === sfn.IntegrationPattern.RUN_JOB) {
+      policyStatements.push(
+        new iam.PolicyStatement({
+          resources: [
+            cdk.Stack.of(this).formatArn({
+              arnFormat: cdk.ArnFormat.SLASH_RESOURCE_SLASH_RESOURCE_NAME,
+              service: 'emr-containers',
+              resource: 'virtualclusters',
+              resourceName: sfn.JsonPath.isEncodedJsonPath(this.props.virtualCluster.id) ? '*' : `${this.props.virtualCluster.id}/jobruns/*`, // Need wild card for dynamic start job run https://docs.aws.amazon.com/step-functions/latest/dg/emr-eks-iam.html
+            }),
+          ],
+          actions: [
+            'emr-containers:DescribeJobRun',
+            'emr-containers:CancelJobRun',
+          ],
+        }),
+      );
+    }
+
+    return policyStatements;
+  }
+}
+
+/**
+ * The information about job driver for Spark submit.
+ */
+export interface SparkSubmitJobDriver {
+
+  /**
+   * The entry point of job application.
+   *
+   * Length Constraints: Minimum length of 1. Maximum length of 256.
+   */
+  readonly entryPoint: sfn.TaskInput;
+
+  /**
+   * The arguments for a job application in a task input object containing an array of strings
+   *
+   * Length Constraints: Minimum length of 1. Maximum length of 10280.
+   * @type string[]
```

Comment:
I feel like the `@type` is confusing here, since `entryPointArguments` is not of type `string[]`.

Reply:
Changed to `@type sfn.TaskInput which expects payload as an array of strings`